pkgs/development/python-modules/visions/default.nix

```diff
@@ -0,0 +1,75 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, pytestCheckHook
+, attrs
+, imagehash
+, matplotlib
+, multimethod
+, networkx
+, numpy
+, pandas
+, pillow
+, pydot
+, pygraphviz
+, shapely
+}:
+
+buildPythonPackage rec {
+  pname = "visions";
+  # latest release (Dec 2021) requires barely used 1.8GB `tangled-up-in-unicode` packae
+  version = "unstable-2022-09-30";
+  disabled = pythonOlder "3.6";
+
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "dylan-profiler";
+    repo = pname;
```

Comment:
```suggestion
    repo = "visions";
```

Reply:
Since `repo` and `pname` are not semantically the same, it is better not to reuse variables.

Reply:
Thanks for the suggestion, but I'd prefer to leave as is for now unless this guideline exists in the Nixpkgs manual somewhere -- the majority of my packages follow the current variable binding structure and I can't remember being challenged by reviewers.  (Also, I'd argue that they are semantically essentially the same, and the alternative `let nameOfThePackage = ...; in mkDerivation { pname = nameOfThePackage; src = fetchFromGitHub { repo = nameOfThePackage }` is worse here.